api/envoy/api/v2/route/route_components.proto

```diff
@@ -1010,6 +1010,17 @@ message RetryPolicy {
 
   // HTTP headers which must be present in the request for retries to be attempted.
   repeated HeaderMatcher retriable_request_headers = 10;
+
+  // The name of the retry policy. The field is used for pluggable retry policy.
+  string name = 11 [(validate.rules).string = {min_bytes: 1}];
+
+  // Implementation specific configuration which depends on the implementation being instantiated.
+  // See the supported retry policy implementations for further documentation.
+  oneof config_type {
+    google.protobuf.Struct config = 12 [deprecated = true];
```

Comment:
For new extensions we can delete this and just provide typed_config

Reply:
It seems that the `translateOpaqueConfig` requires the message to contain `config` field. For v3 proto, it doesn't include this field.

Reply:
See `CacheFilterFactory::createFilterFactoryFromProtoTyped` for how to do this directly in the new style. Perhaps we should have a utility function for this though? cc @kyessenov for guidance.

Reply:
Please remove `config` field. We should use `typed_config` from now on for all extensions and used TypedStruct inside it if necessary.
I think we usually have a short proto for an extension:
```
message PluggableReryPolicy {
  string name = 1;
  google.protobuf.Any typed_config = 2;
}
```
and then add a field in the outer class for it. That is how we define access log and filter extensions, so we could try that here, too.

Reply:
Thanks @kyessenov! Added a utility function and removed `config` field.